src/odfesqlodbc/es_communication.cpp

```diff
@@ -102,18 +102,18 @@ void ESCommunication::AwsHttpResponseToString(
     output.assign(buf.data(), avail);
 }
 
-void ESCommunication::PrepareCursorResult(ESResult& es_result) {
+void ESCommunication::PrepareCursorResult(std::unique_ptr< ESResult >& es_result) {
```

Comment:
Should this be a const ref?

Reply:
Actually - if this is a unique ptr to begin with it'd be better to invoke as: PrepareCursorResult(*es_result)
and keep the input argument as a const ref.

Reply:
const won't work since parse is modifying es_result_doc